samples: boards: nxp: mcxn_a s2ram/system_off fail to build on MCXN EVK targets due to missing wakeup_button alias

﻿### Describe the bug

Building Zephyr samples `sample.boards.nxp.mcxn_a.s2ram.button` and `sample.boards.nxp.mcxn_a.system_off.button` fails on NXP MCXN platforms because `DT_ALIAS(wakeup_button)` does not resolve to a valid alias for the tested boards, causing `WUC_DT_SPEC_GET()` expansion to fail at compile time.

This was observed in upstream-main based CI runs for both:
- `mcx_n5xx_evk/mcxn547/cpu0`
- `mcx_n9xx_evk/mcxn947/cpu0`


### Regression

- [x] This is a regression.

### Steps to reproduce

1. Clone Zephyr main.
2. Check out commit `0a9cc79d760f387df304892f6e0bba92b62adeaf`.
3. Build either failing sample on one of the affected boards, for example:
   - `scripts/twister -j 10 --inline-logs --detailed-test-id -T samples -p mcx_n5xx_evk/mcxn547/cpu0 -W`
   - or `west build -b mcx_n5xx_evk/mcxn547/cpu0 samples/boards/nxp/mcxn_a/s2ram`
   - or `west build -b mcx_n5xx_evk/mcxn547/cpu0 samples/boards/nxp/mcxn_a/system_off`
4. Observe the compile failure from `WUC_DT_SPEC_GET(DT_ALIAS(wakeup_button))`.

The same failure was also seen on `mcx_n9xx_evk/mcxn947/cpu0`.

### Relevant log output

```shell
/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree/wuc.h:173:40: note: in expansion of macro 'DT_PHA_BY_IDX'
  173 | #define DT_WUC_ID_BY_IDX(node_id, idx) DT_PHA_BY_IDX(node_id, wakeup_ctrls, idx, id)
      |                                        ^~~~~~~~~~~~~
/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/drivers/wuc.h:73:67: note: in expansion of macro 'DT_WUC_ID_BY_IDX'
   73 |         {.dev = DEVICE_DT_GET(DT_WUC_BY_IDX(node_id, idx)), .id = DT_WUC_ID_BY_IDX(node_id, idx)}
      |                                                                   ^~~~~~~~~~~~~~~~
/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/drivers/wuc.h:103:34: note: in expansion of macro 'WUC_DT_SPEC_GET_BY_IDX'
  103 | #define WUC_DT_SPEC_GET(node_id) WUC_DT_SPEC_GET_BY_IDX(node_id, 0)
      |                                  ^~~~~~~~~~~~~~~~~~~~~~
/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/samples/boards/nxp/mcxn_a/s2ram/src/main.c:78:42: note: in expansion of macro 'WUC_DT_SPEC_GET'
   78 | static const struct wuc_dt_spec button = WUC_DT_SPEC_GET(DT_ALIAS(wakeup_button));
      |                                          ^~~~~~~~~~~~~~~
/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:288:25: note: in expansion of macro 'DT_CAT'
  288 | #define DT_ALIAS(alias) DT_CAT(DT_N_ALIAS_, alias)
      |                         ^~~~~~
/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/samples/boards/nxp/mcxn_a/s2ram/src/main.c:78:58: note: in expansion of macro 'DT_ALIAS'
   78 | static const struct wuc_dt_spec button = WUC_DT_SPEC_GET(DT_ALIAS(wakeup_button));
      |                                                          ^~~~~~~~
ninja: build stopped: subcommand failed.
```

`system_off` fails in the same way:

```shell
/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/samples/boards/nxp/mcxn_a/system_off/src/main.c:25:42: note: in expansion of macro 'WUC_DT_SPEC_GET'
   25 | static const struct wuc_dt_spec wakeup = WUC_DT_SPEC_GET(DT_ALIAS(wakeup_button));
      |                                          ^~~~~~~~~~~~~~~
...
ninja: build stopped: subcommand failed.
```

### Impact

Major â€“ Affected NXP MCXN board samples do not build in CI on the tested platforms.

### Environment

- OS: Linux (Jenkins CI environment)
- Toolchain: zephyr
- Commit SHA: `0a9cc79d760f387df304892f6e0bba92b62adeaf`
- Zephyr version string from CI: `v4.4.0-8029-g0a9cc79d760f`
- Boards observed:
  - `mcx_n5xx_evk/mcxn547/cpu0`
  - `mcx_n9xx_evk/mcxn947/cpu0`

### Additional Context

CI notifications and extracted logs are available locally from:
- `C:\code\goose_working\sandbox\session-20260713-1122-zephyr-failure`

Observed failing tests/samples:
- `sample.boards.nxp.mcxn_a.s2ram.button`
- `sample.boards.nxp.mcxn_a.system_off.button`

The error suggests these samples assume a `wakeup_button` devicetree alias exists for the affected MCXN board targets, but that alias (or the expected `wakeup_ctrls` phandle data behind it) is not available for these platforms.

--- a/boards/nxp/mcx_nx4x_evk/mcx_nx4x_evk.dtsi
+++ b/boards/nxp/mcx_nx4x_evk/mcx_nx4x_evk.dtsi
@@ -7,6 +7,7 @@
 #include <zephyr/dt-bindings/i2c/i2c.h>
 #include <zephyr/dt-bindings/pwm/pwm.h>
 #include <zephyr/dt-bindings/input/input-event-codes.h>
+#include <zephyr/dt-bindings/wuc/wuc_nxp_wuu.h>
 
 / {
 	aliases {
@@ -15,6 +16,7 @@
 		led2 = &blue_led;
 		sw0 = &user_button_2;
 		sw1 = &user_button_3;
+		wakeup-button = &user_button_2;
 		sdhc0 = &usdhc0;
 		mcuboot-button0 = &user_button_2;
 	};
@@ -65,6 +67,8 @@
 			label = "User SW2";
 			gpios = <&gpio1 3 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>;
 			zephyr,code = <INPUT_KEY_0>;
+			/* SW2 (P1_3) wakes via WUU external pin 7. */
+			wakeup-ctrls = <&wuu NXP_WUU_PIN_7_FALLING_INT>;
 			status = "disabled";
 		};
 

